src/Components/Blazor/WebAssembly.Authentication/src/AuthenticationManager.cs

```diff
@@ -0,0 +1,446 @@
+                    {
+                        return Uri.UnescapeDataString(value.Replace('+', ' '));
+                    }
+
+                    equalIndex = queryString.IndexOf('=', ampersandIndex);
+                    if (equalIndex == -1)
+                    {
+                        equalIndex = textLength;
+                    }
+                }
+                else
+                {
+                    if (ampersandIndex > scanIndex)
+                    {
+                        var value = queryString[scanIndex..ampersandIndex];
+                        if (string.Equals(value, key, StringComparison.OrdinalIgnoreCase))
+                        {
+                            return string.Empty;
+                        }
+                    }
+                }
+
+                scanIndex = ampersandIndex + 1;
+            }
+
+            return null;
+        }
+
+        private async Task NavigateToReturnUrl(string returnUrl) => await JS.InvokeVoidAsync("location.replace", returnUrl);
+
+        private ValueTask RedirectToRegister()
+        {
+            var loginUrl = Navigation.ToAbsoluteUri(ApplicationPaths.LoginPath).PathAndQuery;
+            var registerUrl = Navigation.ToAbsoluteUri($"{ApplicationPaths.RemoteRegisterPath}?returnUrl={loginUrl}").PathAndQuery;
+
+            return JS.InvokeVoidAsync("location.replace", registerUrl);
+        }
+
+        private ValueTask RedirectToProfile() => JS.InvokeVoidAsync("location.replace", Navigation.ToAbsoluteUri(ApplicationPaths.RemoteProfilePath).PathAndQuery);
+
+        private string GetErrorMessage() => GetParameter("message");
+
+        private static void DefaultLoginFragment(RenderTreeBuilder builder)
+        {
+            builder.OpenElement(0, "p");
+            builder.AddContent(1, "Checking login state...");
+            builder.CloseElement();
+        }
+
+        private static void RegisterNotSupportedFragment(RenderTreeBuilder builder)
+        {
+            builder.OpenElement(0, "p");
+            builder.AddContent(1, "Registration is not supported.");
+            builder.CloseElement();
+        }
+
+        private static void ProfileNotSupportedFragment(RenderTreeBuilder builder)
+        {
+            builder.OpenElement(0, "p");
+            builder.AddContent(1, "Editing the profile is not supported.");
+            builder.CloseElement();
+        }
+
+        private static void DefaultLoginCallbackFragment(RenderTreeBuilder builder)
+        {
+            builder.OpenElement(0, "p");
+            builder.AddContent(1, "Completing login...");
+            builder.CloseElement();
+        }
+
+        private static RenderFragment DefaultLoginFailedFragment(string message)
+        {
+            return builder =>
+            {
+                builder.OpenElement(0, "p");
+                builder.AddContent(1, "There was an error trying to log you in: '");
+                builder.AddContent(2, message);
```

Comment:
Just to check: is the message we have here definitely intended for end users to see? If it's going to be something like "login was cancelled" then this makes sense, though does have some pretty bad localisation problems.

On the other hand, if it's something like an error code or a technical description of a misconfiguration, then it seems like something to log instead, and just display a more generic human-readable message.

Reply:
OK thanks